core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -16,203 +16,246 @@
+    const validateExplorationsUrl =
+      this.urlInterpolationService.interpolateUrl(
+        StoryDomainConstants.VALIDATE_EXPLORATIONS_URL_TEMPLATE, {
           story_id: storyId
         });
 
-      $http.get(validateExplorationsUrl, {
+    this.http.get<void>(
+      validateExplorationsUrl, {
+        observe: 'response',
         params: {
           comma_separated_exp_ids: expIds.join(',')
         }
-      }).then(function(response) {
-        if (successCallback) {
-          successCallback(response.data.validation_error_messages);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+      }
+    ).toPromise().then(
+      response => successCallback(response.status),
+      errorResponse => errorCallback(errorResponse.error.error));
+  }
+
+  private _deleteStory(
+      storyId: string,
+      successCallback: (value:number) =>void,
+      errorCallback: (reason: string) => void): void {
+    const storyDataUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.EDITABLE_STORY_DATA_URL_TEMPLATE, {
+        story_id: storyId
       });
-    };
 
-    var _deleteStory = function(
-        storyId, successCallback, errorCallback) {
-      var storyDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_STORY_DATA_URL_TEMPLATE, {
-          story_id: storyId
-        });
-      $http['delete'](storyDataUrl).then(function(response) {
+    // eslint-disable-next-line
+    this.http.delete(
+      storyDataUrl, {observe: 'response'}).toPromise().then(
+      (response) => {
         if (successCallback) {
           successCallback(response.status);
         }
-      }, function(errorResponse) {
+      }, (errorResponse) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(errorResponse.error.error);
         }
       });
-    };
+  }
 
-    var _doesStoryWithUrlFragmentExist = function(
-        storyUrlFragment, successCallback, errorCallback) {
-      var storyUrlFragmentUrl = UrlInterpolationService.interpolateUrl(
-        STORY_URL_FRAGMENT_HANDLER_URL_TEMPLATE, {
-          story_url_fragment: storyUrlFragment
-        });
-      $http.get(storyUrlFragmentUrl).then(function(response) {
+  private _doesStoryWithUrlFragmentExist(
+      storyUrlFragment: string,
+      successCallback: (value: StoryUrlFragmentExists) => void,
+      errorCallback: (reason: string) => void): void {
+    const storyUrlFragmentUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.STORY_URL_FRAGMENT_HANDLER_URL_TEMPLATE, {
+        story_url_fragment: storyUrlFragment
+      });
+    this.http.get<StoryUrlFragmentExists>(
+      storyUrlFragmentUrl).toPromise().then(
+      (response) => {
         if (successCallback) {
-          successCallback(response.data.story_url_fragment_exists);
+          successCallback(response.storyUrlFragmentExists);
```

Comment:
Shouldn't it be `response.story_url_fragment_exists` ?

Reply:
corrected!